src/coordinax/_src/vectors/checks.py

```diff
@@ -82,3 +78,179 @@ def check_polar_range(
         xp.any(xp.logical_or((polar < _l), (polar > _u))),
         "The inclination angle must be in the range [0, pi].",
     )
+
+
+def check_non_negative(x: AbstractQuantity, /, name: str = "") -> AbstractQuantity:
+    """Check that the input is non-negative.
+
+    Examples
+    --------
+    >>> from unxt import Quantity
+
+    Pass through the input if the value is non-negative.
+
+    >>> x = Quantity([0, 1, 2], "m")
+    >>> check_non_negative(x)
+    Quantity['length'](Array([0, 1, 2], dtype=int32), unit='m')
+
+    Raise an error if any value is negative.
+
+    >>> x = Quantity([-1, 1, 2], "m")
+    >>> try: check_non_negative(x)
+    ... except Exception: pass
+
+    """
+    name = f" {name}" if name else name
+    return eqx.error_if(x, xp.any(x < 0), f"The input{name} must be non-negative.")
+
+
+def check_non_negative_non_zero(
+    x: AbstractQuantity, /, name: str = ""
+) -> AbstractQuantity:
+    """Check that the input is non-negative and non-zero.
+
+    Examples
+    --------
+    >>> from unxt import Quantity
+
+    Pass through the input if the value is non-negative.
+
+    >>> x = Quantity([1, 2, 3], "m")
+    >>> check_non_negative_non_zero(x)
+    Quantity['length'](Array([1, 2, 3], dtype=int32), unit='m')
+
+    Raise an error if any value is negative or zero.
+
+    >>> x = Quantity([-1, 1, 2], "m")
+    >>> try: check_non_negative_non_zero(x)
+    ... except Exception: pass
+
+    >>> x = Quantity([0, 1, 2], "m")
+    >>> try: check_non_negative_non_zero(x)
+    ... except Exception: pass
+
+    """
+    name = f" {name}" if name else name
+    return eqx.error_if(
+        x, xp.any(x <= 0), f"The input{name} must be non-negative and non-zero."
+    )
+
+
+def check_less_than(
+    x: AbstractQuantity,
+    max_val: AbstractQuantity,
+    /,
+    name: str = "",
+    comparison_name: str = "the specified maximum value",
```

Comment:
Can this be more verbose while retaining clarity?

Reply:
ChatGPT suggests:

•	ref_name (short for “reference name”)
•	limit_name (indicating the name of the limiting value)
•	max_name (if it’s always describing the maximum)
•	comp_name (short for “comparison name”)

Which do you prefer / have another suggestion?

Reply:
🤷 - is it worth it for a few characters on an internal function? I can, but personally don't think it's worth it!